### PR TITLE
PYTHON-2793 Stop testing TLS with Python 3.10 and MongoDB <= 3.4

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2431,6 +2431,20 @@ buildvariants:
   display_name: "PyOpenSSL ${platform} ${python-version-windows} ${auth}"
   tasks:
     - '.replica_set'
+  rules:
+    # Python 3.10 does not support TLS on MongoDB <= 3.4.
```
Comment (behackett) [inline .evergreen/config.yml:2435]: It's more like "On Windows, Python 3.10+ and MongoDB <= 3.4 don't have any ciphers in common."

File: .evergreen/config.yml [hunk 2]
```diff
@@ -2431,6 +2431,20 @@ buildvariants:
   display_name: "PyOpenSSL ${platform} ${python-version-windows} ${auth}"
   tasks:
     - '.replica_set'
+  rules:
```
Comment (behackett) [inline .evergreen/config.yml:2434]: Is this actually a problem when using PyOpenSSL, or is it only a problem with CPython's ssl module?

File: .evergreen/config.yml [hunk 3]
```diff
@@ -2533,6 +2547,18 @@ buildvariants:
     auth-ssl: "*"
   display_name: "${platform} ${python-version-windows} ${auth-ssl}"
   tasks: *all-server-versions
+  rules:
+    # Python 3.10 does not support TLS on MongoDB <= 3.4.
+    - if:
+        platform: "*"
+        python-version-windows: "3.10"
+        auth-ssl: "auth-ssl"
```
Comment (behackett) [inline .evergreen/config.yml:2555]: I think we still want to test auth, right? Doesn't this skip tls and auth?

File: .evergreen/config.yml [hunk 4]
```diff
@@ -2541,6 +2567,18 @@ buildvariants:
     auth-ssl: "*"
   display_name: "${platform} ${python-version-windows-32} ${auth-ssl}"
   tasks: *all-server-versions
+  rules:
+    # Python 3.10 does not support TLS on MongoDB <= 3.4.
+    - if:
+        platform: "*"
+        python-version-windows-32: "3.10"
+        auth-ssl: "auth-ssl"
```
Comment (behackett) [inline .evergreen/config.yml:2575]: Same comment here.

File: doc/examples/tls.rst [hunk 5]
```diff
@@ -249,3 +249,14 @@ revocation checking failed::
   [('SSL routines', 'tls_process_initial_server_flight', 'invalid status response')]
 
 See :ref:`OCSP` for more details.
+
+Python 3.10+ is incompatible with TLS/SSL on MongoDB <= 3.4
```
Comment (behackett) [inline doc/examples/tls.rst:253]: On Windows. It seems to work just fine everywhere else.

File: doc/examples/tls.rst [hunk 6]
```diff
@@ -249,3 +249,14 @@ revocation checking failed::
...
+...........................................................
+
+Note that Python 3.10+ is incompatible with TLS/SSL on MongoDB <= 3.4. The
+following are some example errors that may occur with this combination::
+
+  SSL handshake failed: localhost:27017: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:997)
```
Comment (behackett) [inline doc/examples/tls.rst:259]: I think there is also an error about cipher mismatch, which is the real problem.

Comment (behackett) [reply doc/examples/tls.rst:259]: From the ticket description, we have this in the server log:
```SSL: error:1408A0C1:SSL routines:ssl3_get_client_hello:no shared cipher```

File: .evergreen/config.yml (hunk 1 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml]: Done.

File: .evergreen/config.yml (hunk 2 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml]: Oh right, I was confused by the other test failure. PyOpenSSL+3.10+MongoDB 2.6 works fine.

File: doc/examples/tls.rst (hunk 6 above)
Comment (ShaneHarvey) [reply doc/examples/tls.rst:259]: Done.

File: .evergreen/config.yml (hunk 3 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml:2541]: Done.

File: .evergreen/config.yml (hunk 4 above)
Comment (ShaneHarvey) [reply .evergreen/config.yml:2574]: Done.

File: doc/examples/tls.rst (hunk 5 above)
Comment (ShaneHarvey) [reply doc/examples/tls.rst:253]: I tested locally on macos and it's the same error behavior due to no shared ciphers.

Comment (behackett) [reply doc/examples/tls.rst:253]: Thanks for checking.

**Review by behackett — APPROVED** [review]

**Pull request merged**